iree/turbine/kernel/wave/analysis/index_sequence_analysis.py

```diff
@@ -525,6 +541,128 @@ def set_thread_dependent_index_from_read_write(
         )
 
 
+def get_reduce_mapping(
+    trace: CapturedTrace, constraints: list[Constraint]
+) -> dict[ReduceOp, dict[IndexSymbol, IndexSequence]]:
+    """
+    Get the mapping of the reduce ops to the index sequence.
+    """
+    sources = trace.walk(lambda node: isinstance(get_custom(node), ReduceOp))
+    hardware_constraint = get_hardware_constraint(constraints)
+    workgroup_constraints = [
```

Comment:
minor: can you use get_workgroup_constraints in general_utils ?

Reply:
done